src/analyze/analyze.c

```diff
@@ -1275,36 +1276,87 @@ static int set_log_target(sd_bus *bus, char **args) {
         return 0;
 }
 
+static int dump_seccomp_filter(const char *name) {
+        const SyscallFilterSet *set;
+        const char *syscall;
+
+        set = syscall_filter_set_find(name);
+        if (!set) {
+                log_error("Filter set \"%s\" not found.", name);
+                return -ENOENT;
+        }
+
+        printf("%s\n", set->name);
+        NULSTR_FOREACH(syscall, set->value)
+                printf("    %s\n", syscall);
+
+        return 0;
+}
+
+static int dump_seccomp_filters(char** names) {
+        bool first = true;
+        int r;
+
+        pager_open(arg_no_pager, false);
+
+        if (strv_isempty(names)) {
+                int i;
+
+                for (i = 0; i < _SYSCALL_FILTER_SET_MAX; i++) {
+                        if (!first)
+                                puts("");
+                        r = dump_seccomp_filter(syscall_filter_sets[i].name);
+                        if (r < 0)
+                                return r;
+                        first = false;
+                }
```

Comment:
not that it matters given the sizes, but this results in O(n^2) complexity, since syscall_filter_set_find() is linear. I'd probably change this minimally, and pass the SyscallFilterSet object directly to dump_seccomp_filter(), to avoid this... (and then keep a version that takes the name as a wrapper around it called dump_seccomp_filter_by_name() or so?)